lib/ansible/module_utils/yumdnf.py

```diff
@@ -98,6 +98,15 @@ def __init__(self, module):
         self.enablerepo = self.listify_comma_sep_strings_in_list(self.enablerepo)
         self.exclude = self.listify_comma_sep_strings_in_list(self.exclude)
 
+        # Fail if someone passed a space separated string
+        # https://github.com/ansible/ansible/issues/46301
+        if any((' ' in name for name in self.names)):
```

Comment:
Hrm, so my assumption "who would put space in a package name?" wasn't true of course. This breaks `@Development Tools` use case.

Reply:
Right, so a rpm package name can *not* contain a space, but groups can. I guess you can just add a `and '@' not in name` to your conditional and that should handle it.